wiki_articles/windows_msys2_gcc.md

```diff
@@ -0,0 +1,35 @@
+# How to install msys2 and gcc on windows
+In this guide, we install msys2 and gcc. There are some optional tools that we recommend
+installing, such as `clangd` and `clang-format`. Having an alternative compiler, `clang`
+is also recommended. Install location is assumed to be the default.
+
+<!-- inline -->
+## Msys2 and GCC Toolchain
+1. Visit [msys2.org](https://msys2.org) and follow their installation guide upto step 4.
+2. After installing, open the Msys2 Msys terminal environment on windows. It is an app.
```

Comment:
I think it'd be best to have them add C:\msys64\usr\bin and C:\msys64\ucrt64\bin to the PATH during step 2. Msys' shells are really weird and I think it's will be really unintuitive for beginners. It'd be best to just have them be able to do everything through windows terminal / vscode's terminal which are both normal powershell. I'd like to avoid anything like the "Use the `Msys ucrt64` app if you want access to these tools directly." note if at all possible.

Reply:
> I think it'd be best to have them add C:\msys64\usr\bin and C:\msys64\ucrt64\bin to the PATH during step 2. Msys' shells are really weird and I think it's will be really unintuitive for beginners. It'd be best to just have them be able to do everything through windows terminal / vscode's terminal which are both normal powershell.

I strongly think that's not really required. We're writing a brief, to-the-point tutorial article and we can expect some basic know how from our readers that they they can open an App from the start menu. I feel it deviates from a typical setup, if you're installing gcc, then you better have it installed first. Fiddle with paths later.

>  I'd like to avoid anything like the "Use the Msys ucrt64 app if you want access to these tools directly." note if at all possible.
Removed/changed the language because that wasn't the intended purpose. It was merely an informative note.

Reply:
1. You can do everything from MSYS2 UCRT64 terminal. This is slightly easier than telling people to open MSYS2 MSYS for running `pacman`.

2. I personally wouldn't add `C:\msys64\usr\bin` to PATH, to avoid exposing cygwin utilities. But if you do add it, it MUST be after `C:\msys64\ucrt64\bin`.

3. If you decide to recommend modifying PATH, we should recommend adding `C:\msys64\ucrt64\bin` to the beginning of the system-wide PATH, for it to get priority over everything else.

Reply:
Yeah I would also not recommend to add stuff to PATH. That's generally not how things are supposed to be done on Windows, and we regularly see people run into problems with that with MinGW in particular. Other Software not really designed for Windows (notable example: neovim) will add its own stuff to PATH by default. And then it becomes a matter of which one happened to add itself earlier in the list to determine whether it's neovim that will crash, or your programs compiled with MinGW (because there are not multiple libstdc++ dlls on the PATH and it's a matter of which one appears on PATH first to determine which one will end up using the wrong one).

Reply:
@michael-kenzel Adding MinGW to the beginning of PATH should be safe. It shouldn't cause any crashes in other software because loading DLLs from the same directory as the .exe requesting them has priority over PATH, and the only executables with no DLLs in the same directory should be the ones you compile yourself.

Reply:
@michael-kenzel Adding MinGW to the beginning of PATH should be safe. It shouldn't cause any crashes in other software because loading DLLs from the same directory as the .exe requesting them has priority over PATH, and the only executables with no DLLs in the same directory should be the ones you compile yourself.

Reply:
The main reason I recommend adding msys stuff to the PATH is to stem inevitable confusion about why commands only work in the msys terminal and not normal cmd/powershell. This is especially relevant if the user eventually wants to use vscode.

Reply:
I just checked, neovim (at least the build you get via winget) doesn't seem to actually add itself to PATH. So I guess the people who were using issues were using some other build. It's still not the greatest idea to add directories that may contain widely used standard dependencies to PATH, because the issues you get as a result are pretty much impossible for a beginner to diagnose.